include/PetriEngine/Structures/PotencyQueue.h

```diff
@@ -69,6 +69,32 @@ namespace PetriEngine {
         private:
             size_t _seed;
         };
+
+        class RandomWalkPotencyQueue : public PotencyQueue {

```

Comment:
As I use the same function push and the same attribute _seed, maybe inherit from RandomPotensyQueue instead of PotencyQueue

Reply:
Only inherit if it actually makes sense & if it has significant code-use. Otherwise it create un-needed dependencies in the code-base.

Reply:
I think it makes sense, because a random walk queue (based on potency) *is* a random potency queue. It uses all the functions of RandomPotencyQueue except pop that have a different behaviour.